packages/blade/src/components/Collapsible/CollapsiblePanel.web.tsx

```diff
@@ -0,0 +1,127 @@
+import type { ReactElement, ReactNode, TransitionEventHandler } from 'react';
+import { useRef } from 'react';
+import styled from 'styled-components';
+import { useCollapsibleContext } from './CollapsibleContext';
+import { castWebType, makeMotionTime, makeSize } from '~utils';
+import { useDidUpdate } from '~src/hooks/useDidUpdate';
+
+type CollapsiblePanelProps = {
+  children: ReactNode;
+};
+
+type StyledCollapsiblePanelProps = {
+  defaultIsExpanded: boolean;
+  isExpanded: boolean;
+};
+
+/**
+ * We can't animate to and from `auto` height, these are used for imperative css transitions.
+ * Overall on expanding height will change as: 0px -> Actual height -> auto
+ */
+const HEIGHT_EXPANDED = 'auto';
+const HEIGHT_COLLAPSED = '0px';
+
+// TODO: move common styles when implementing native
+const StyledCollapsiblePanel = styled.div<StyledCollapsiblePanelProps>((props) => {
+  const { theme, defaultIsExpanded, isExpanded } = props;
+  return {
+    transitionDuration: castWebType(makeMotionTime(theme.motion.duration.xmoderate)),
+    transitionTimingFunction: castWebType(theme.motion.easing.standard.effective),
+    transitionProperty: 'height, opacity',
+    opacity: isExpanded ? 1 : 0.8,
+    /**
+     * We need height explicitly here for initial styles because the component might be rendered on server,
+     * in which case for expanded items this should be `auto` because we don't know the actual pixel value.
+     *
+     * This is based on `defaultIsExpanded` rather than `isExpanded` because post initial render inline styles take over.
+     * Otherwise, changing `height` both here and in inline styles sometimes causes animation flickers due to styles mismatch.
+     */
+    height: defaultIsExpanded ? HEIGHT_EXPANDED : HEIGHT_COLLAPSED,
+    display: defaultIsExpanded ? 'block' : 'none',
+    overflowY: 'hidden',
```

Comment:
why this?

Reply:
This is needed because the content will occupy its own height even while the content expands / collapses, the overflow ensures that the overflowing content isn't visible.

Eg. consider content might be 200px in height, say the panel is collapsing at time `t` where its height is 50px we would need the overflow so the remaining 150px of content isn't visible.